src/util/setSelection.ts

```diff
@@ -5,20 +5,23 @@
   @param offset    Character offset of selection.
   @param end       If true, sets the offset to the end of the text.
  */
-export const setSelection = (el, { offset, end } = {}) => {
+type SelectionOptionsType = {offset: number, end: number}
+export const setSelection = (el: HTMLElement, { offset, end }: SelectionOptionsType = {offset: 0, end: 0}) => {
   if (el.childNodes.length === 0) {
     el.appendChild(document.createTextNode(''))
   }
   const textNode = el.firstChild
   const range = document.createRange()
-  const sel = window.getSelection()
+  const sel = window.getSelection() || new Selection()
```

Comment:
Can you explain this?

Reply:
When used in Firefox, `window.getSelection()` [returns null](https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection) for some values, and hence, TS compilation errors on [these lines](https://github.com/cybersemics/em/blob/094828d323f63e1d19d865947e619322c9dc2519/src/util/setSelection.ts#L27)